test/blackbox-tests/test-cases/merlin/github4125.t/run.t

```diff
@@ -0,0 +1,39 @@
+  $ unset INSIDE_DUNE
+  $ dune build
```

Comment:
Something doesn't seem right here. If you unset `INSIDE_DUNE`, then `dune build` is going to:

- read the user configuration file
- lookup the root outside of the test
- and a few other things

Which is going to make it difficult to keep this test reproducible.

Why do you need to unset `INSIDE_DUNE`?

Reply:
Because without it the Dune in the test file doesn't tries to reach the root of the test project and thus the test where I launch `dune ocaml-merlin` from a subfolder fails because the `dune-workspace` file at the root of the project is not loaded. It is a test about root discovery...

With `INDISE_DUNE` unset in the test dune does find the correct root for the test, that is `$TESTCASE_ROOT` and not the root of the real dune project.

Other tests already being at the root pass without problem so i can just remove this one if you prefer.

https://github.com/ocaml/dune/blob/bebb7a9432b149fce95ac820757216372afb8f0f/bin/workspace_root.ml#L90-L93

Reply:
> With INSIDE_DUNE unset in the test dune does find the correct root for the test, that is $TESTCASE_ROOT and not the root of the real dune project.

This is because `dune-workspace` has precedence over `dune-project`. If you add a `dune-workspace` file at the root of dune, then the test will pick this as root.

If we want to test this properly, we need a way to tell Dune to stop the search at `$TESTCASE_ROOT`. But unsetting `INSIDE_DUNE` seems too dodgy.

Reply:
For now I removed the test in question.